scripts/quests/ahtUrhgan/The_Prince_and_the_Hopper.lua

```diff
@@ -123,19 +132,32 @@ quest.sections =
                 end,
             },
 
+            onZoneIn = {
+                function(player, prevZone)
+                    if quest:getVar(player, 'Prog') == 4 then
+                        return 227
+                    end
+                end,
+            },
+
             onEventFinish =
             {
                 [222] = function(player, csid, option, npc)
                     quest:setVar(player, 'Prog', 3)
-                    return quest:event(227)
+                    -- zone the person to wajoam for second part of cs
```

Comment:
Returning event 227 accomplishes the same task, and its not necessary to move the player for this CS to be displayed.